plugins/woocommerce/client/blocks/assets/js/blocks/product-description/edit.tsx

```diff
@@ -124,33 +124,14 @@ function EditableContent( { context = {} } ) {
 }
 
 function Content( props ) {
-	const { context: { queryId, postType, postId } = {}, layoutClassNames } =
-		props;
-	const blockProps = useBlockProps( { className: layoutClassNames } );
+	const { context: { postType, postId } = {}, layoutClassNames } = props;
 
 	const userCanEdit = useCanEditEntity( 'postType', postType, postId );
 	if ( userCanEdit === undefined ) {
 		return null;
 	}
 
-	// Show warning if the block is not used in a product context.
-	if ( postType && postType !== 'product' ) {
-		return (
-			<div { ...blockProps }>
-				<Warning>
-					{ __(
-						'This block requires a product context to function. You can use it inside a Single Product template, Single Product block, Product Collection block, or within other product-related blocks.',
-						'woocommerce'
-					) }
-				</Warning>
-			</div>
-		);
-	}
-
-	const isDescendentOfProductCollection = Number.isFinite( queryId );
-	const isEditable = userCanEdit && ! isDescendentOfProductCollection;
-
-	return isEditable ? (
+	return userCanEdit ? (
```

Comment:
I think we should add the product collection limitation back in 🙈 sorry!
In testing it, I noticed that you can click on each block within a query loop for each individual product/post that is rendered, but it will select the same block type in the block list view.
This causes some interesting behaviour around blocks that have a block based descriptions and one's that don't, where the context keeps switching ( see video below ). 
This gives me the sense that the blocks shown within a query loop should ideally not be content aware for handling any settings. If we add this edit support, it would mean they are content aware and users see different things depending on which product they are looking at:

https://github.com/user-attachments/assets/0ee3322c-3f53-4814-97d8-332e2d8888da

Reply:
I am not sure if I understand you correctly because I still think the behavior is as expected. Basically, when there are no inner blocks, we have this classic inner block that allows us to edit the product description. However, if there are inner blocks like paragraphs, users can make changes to those paragraphs directly, and after the user saves the post, the product description will also be updated. So, the editing of the product description still works. I am uncertain if I am understanding you correctly. Isn't that what we want? Here is a quick demo video:

https://github.com/user-attachments/assets/4e45e379-671b-417e-bc5e-4e2997a480af

Reply:
Aaah I didn't notice that it does explicitly list the changed products on save, that is pretty cool!

Having said that, I still wonder if there is a larger reason for having this condition ( I asked in Slack to see ), because the post content block is not the only one the condition is also in the:
- Post author
- Post date
- Post title
- Post excerpt
- Post Image